### PR TITLE
Update Maven project

### PR DESCRIPTION
**Pull request by edewata**
New `pom.xml` files have been added to define Maven modules for PKI. The dependencies have also been updated to match the available packages on Fedora.

The build test has been updated to build PKI with Maven and compare the build artifacts with the original CMake build artifacts.

**Note:** Some of Maven build artifacts (e.g. `pki-server.jar`) do not have matching CMake build artifacts (e.g. `pki-cms.jar`, `pki-cmsbundle.jar`) so they cannot be compared directly. This will be addressed in the future.

Azure pipeline:
https://dev.azure.com/dogtagpki/pki/_build?definitionId=2

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,157 +6,222 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>org.dogtagpki</groupId>
     <artifactId>pki</artifactId>
-    <version>11.0.0-SNAPSHOT</version>
+    <version>11.3.0-SNAPSHOT</version>
```
Comment (fmarco76) [inline pom.xml:9]: This version is repeated in all pom files. I think it is better to define a propertiy section [1] here and use it everywhere to simplify the management of the versions.
Other approaches are possible [2].

[1] https://maven.apache.org/pom.html#Properties
[2] https://stackoverflow.com/questions/17312681/maven-multi-module-project-version-management

**Review by fmarco76 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (ckelleyRH) [reply pom.xml:9]: Agreed!

Comment (edewata) [reply pom.xml]: So apparently there are different ways to manage the versions. One of the recommended methods is actually to use `mvn versions:set` to hard-code the versions into all `pom.xml` files just like in this PR originally, but I've updated the PR to use properties described here: https://maven.apache.org/maven-ci-friendly.html. Could you take a look again? Thanks!

File: base/tomcat-9.0/pom.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    <artifactId>pki-tomcat-9.0</artifactId>
+    <packaging>jar</packaging>
+
+    <dependencies>
+
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>pki-tomcat</artifactId>
+            <version>${project.version}</version>
```
Comment (fmarco76) [inline base/tomcat-9.0/pom.xml:22]: Line 11 and 22 use `revision` and `project.version` respectively. Actually they have the same value but could they diverge in the future? (The same applies to all module `pom.xml`)

**Review by fmarco76 — APPROVED** [review]
LGTM

File: base/tomcat-9.0/pom.xml (hunk 2 above)
Comment (ckelleyRH) [reply base/tomcat-9.0/pom.xml:22]: I think this is fine, I don't think we want to start tracking versions at the module level. Life is already complicated enough with the web of versions we have already!

**Review by ckelleyRH — APPROVED** [review]
LGTM, still trying to think of a way to extract the dep versions into some easily maintainable place but not had a good idea yet.

File: base/tomcat-9.0/pom.xml (hunk 2 above)
Comment (edewata) [reply base/tomcat-9.0/pom.xml:22]: I think this is just one of Maven's quirkiness that we will find as we use it more. Those properties are supposed to be identical (we want to use the same version for all modules), but for the module itself we're supposed to use `revision` (https://maven.apache.org/maven-ci-friendly.html#multi-module-setup), and for the dependencies we're supposed to use `project.version` (https://maven.apache.org/maven-ci-friendly.html#dependencies).

**Comment by edewata** [comment]
@fmarco76 @ckelleyRH Thanks! I'll merge it now. We'll continue to make improvements in this area.

**Pull request merged**